pkg/controller/routes.go

```diff
@@ -72,6 +72,7 @@ func (c *Controller) setupRoutes() error {
 	root.Type(&v1.Workflow{}).HandlerFunc(workflow.EnsureIDs)
 	root.Type(&v1.Workflow{}).HandlerFunc(workflow.CreateWorkspaceAndKnowledgeSet)
 	root.Type(&v1.Workflow{}).HandlerFunc(workflow.BackPopulateAuthStatus)
+	root.Type(&v1.Workflow{}).HandlerFunc(setWorkflowAdditionalCredentialContexts)
```

Comment:
I would be OK with this being a startup migration rather than a controller. What are your thoughts?

Reply:
Good call. That way it doesn't run on every change.